neo/Consensus/ConsensusService.cs

```diff
@@ -223,6 +249,12 @@ private void OnConsensusPayload(ConsensusPayload payload)
                 case ConsensusMessageType.PrepareResponse:
                     OnPrepareResponseReceived(payload, (PrepareResponse)message);
                     break;
+                case ConsensusMessageType.CommitAgreement:
+                    OnCommitAgreement(payload, (CommitAgreement)message);
+                    break;
+                case ConsensusMessageType.Renegeration:
```

Comment:
```
case ConsensusMessageType.Renegeration:
         OnRenegeration(payload, (Renegeration)message);
         break;
```
This will never be reached for a CN which sent out changeView request, and then receives renegeration message from another CN which is in commit phase. Because view numbers are not equal in line 238:
```
if (message.ViewNumber != context.ViewNumber && message.Type != ConsensusMessageType.ChangeView)
```
So line 238, maybe we should check this as well
```
message.Type != ConsensusMessageType.Renegeration
```

Reply:
Onh man, you are right.
As mentioned above, this Regeneration strategy was not tested in practice. This will be our last experiment when everything is ready.

 Thanks for seeing this! Eagle eyes.
Updated to `if (message.ViewNumber != context.ViewNumber && message.Type != ConsensusMessageType.ChangeView && message.Type != ConsensusMessageType.Renegeration)`

Reply:
In fact, it should be:
```cs
if (message.ViewNumber != context.ViewNumber && (message.Type != ConsensusMessageType.ChangeView || message.Type != ConsensusMessageType.Renegeration))
```
I will update.